provider/bikeshed/src/units/essentials.rs

```diff
@@ -18,6 +19,57 @@ use zerovec::ZeroMap;
 
 impl DataProvider<UnitsEssentialsV1Marker> for DatagenProvider {
     fn load(&self, req: DataRequest) -> Result<DataResponse<UnitsEssentialsV1Marker>, DataError> {
+        /// Fills the prefixes map with the SI prefixes (binary and decimal)
+        ///
+        /// # Errors
+        ///
+        /// - `DataError::custom("Invalid prefix: must be Binary or Decimal")` if the prefix is not Binary or Decimal
+        /// - `DataError::custom("Failed to parse pattern key value")` if the pattern key value cannot be parsed
+        fn fill_si(
+            data: &BTreeMap<String, Patterns>,
+            prefixes_map: &mut BTreeMap<PatternKey, String>,
+        ) -> Result<(), DataError> {
+            const BINARY_PREFIX: &str = "1024p";
+            const DECIMAL_PREFIX: &str = "10p";
+
+            data.iter()
+                .filter(|(key, _)| {
+                    key.starts_with(BINARY_PREFIX) || key.starts_with(DECIMAL_PREFIX)
+                })
+                .map(|(key, patterns)| {
+                    let pattern_key_result =
```

Comment:
issue: is this actually a `Result`? There's a `?` in line 54

Reply:
yes, because I want to return `pattern_key_result` as `PatternKey` and as `Result<>`

Reply:
ok you renamed to `pattern_key`